[media-library] asset.getString() throws IllegalStateException

## 🐛 Bug Report
### Summary of Issue <!-- (just a few sentences) -->
In media-library 9.1.0 when asset.getString() called in GetAlbums.kt app crashes and throws an IllegalStateException. It wasn't happened in emulator, i found about this crash when i tested in a device (Samsung galaxy M20 API29)

### Environment - output of `expo diagnostics` & the platform(s) you're targeting
Bare react native project
"react-native": "0.63.2"
"expo-av": "^8.5.0",
"expo-media-library": "^9.1.0",
"react-native-unimodules": "^0.10.1",

buildToolsVersion = "29.0.0"
minSdkVersion = 21
compileSdkVersion = 29
targetSdkVersion = 29

I debugged it in android studio. This is happening because in GetAlbums.kt `asset.getString(bucketDisplayNameIndex)` [called](https://github.com/expo/expo/blob/714a594adb16175b6490ad488f3a97c7fbf922b1/packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/GetAlbums.kt#L42) and implementation of `getString()` in `Cursor.java` says;
```
The result and whether this method **throws an exception** when the column value is null or the column type is not a string type is implementation-defined.
```
So we just need to handle this exception and give it a fallback string to get rid of this error.

When i changed the following line to a normal string in GetAlbums.kt, media-library works as expected.

```
while (asset.moveToNext()) {
            val id = asset.getString(bucketIdIndex)
            val album = albums[id] ?: let {
              Album(
                id = id,
                title = asset.getString(bucketDisplayNameIndex) // i changed this line to title = "example" and error resolves.
              ).also {
                albums[id] = it
              }
            }

            album.count++
          }
```
Thanks!

edit: stack trace added

```
I/System.out: 1 // this is what println(bucketDisplayNameIndex) gives
E/AndroidRuntime: FATAL EXCEPTION: AsyncTask #3
    Process: com.framebyframe, PID: 9371
    java.lang.RuntimeException: An error occurred while executing doInBackground()
        at android.os.AsyncTask$4.done(AsyncTask.java:399)
        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
        at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
        at java.util.concurrent.FutureTask.run(FutureTask.java:271)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
     Caused by: java.lang.IllegalStateException: asset.getString(bucketDisplayNameIndex) must not be null
        at expo.modules.medialibrary.GetAlbums.doInBackground(GetAlbums.kt:42)
        at expo.modules.medialibrary.GetAlbums.doInBackground(GetAlbums.kt:10)
        at android.os.AsyncTask$3.call(AsyncTask.java:378)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:919) 
```

--- a/docs/pages/versions/unversioned/sdk/media-library.md
+++ b/docs/pages/versions/unversioned/sdk/media-library.md
@@ -137,7 +137,7 @@ Queries for user-created albums in media gallery.
 
 #### Returns
 
-An array of [albums](#album).
+An array of [albums](#album). Depending on Android version, root directory of your storage may be listed as album titled _"0"_ or unlisted at all.
 
 ### `MediaLibrary.getAlbumAsync(albumName)`
 

--- a/packages/expo-media-library/CHANGELOG.md
+++ b/packages/expo-media-library/CHANGELOG.md
@@ -8,6 +8,8 @@
 
 ### 🐛 Bug fixes
 
+- Fixed handling albums without name on Android. ([#9787](https://github.com/expo/expo/pull/9787) by [@barthap](https://github.com/barthap))
+
 ## 9.1.0 — 2020-08-13
 
 ### 🎉 New features

--- a/packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/GetAlbums.kt
+++ b/packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/GetAlbums.kt
@@ -1,6 +1,7 @@
 package expo.modules.medialibrary
 
 import android.content.Context
+import android.database.Cursor.FIELD_TYPE_NULL
 import android.os.AsyncTask
 import android.os.Bundle
 import android.provider.MediaStore
@@ -36,14 +37,16 @@ internal open class GetAlbums(private val mContext: Context, private val mPromis
           while (asset.moveToNext()) {
             val id = asset.getString(bucketIdIndex)
 
-            val album = albums[id] ?: let {
-              Album(
+            if (asset.getType(bucketDisplayNameIndex) == FIELD_TYPE_NULL) {
+              continue
+            }
+
+            val album = albums[id] ?: Album(
                 id = id,
                 title = asset.getString(bucketDisplayNameIndex)
               ).also {
                 albums[id] = it
               }
-            }
 
             album.count++
           }
@@ -53,7 +56,7 @@ internal open class GetAlbums(private val mContext: Context, private val mPromis
     } catch (e: SecurityException) {
       mPromise.reject(MediaLibraryConstants.ERROR_UNABLE_TO_LOAD_PERMISSION,
         "Could not get albums: need READ_EXTERNAL_STORAGE permission.", e)
-    } catch (e: IllegalArgumentException) {
+    } catch (e: RuntimeException) {
       mPromise.reject(MediaLibraryConstants.ERROR_UNABLE_TO_LOAD, "Could not get albums.", e)
     }
     return null

--- a/packages/expo-media-library/android/src/test/java/expo/modules/medialibrary/GetAlbumsTests.kt
+++ b/packages/expo-media-library/android/src/test/java/expo/modules/medialibrary/GetAlbumsTests.kt
@@ -2,7 +2,7 @@ package expo.modules.medialibrary
 
 import android.content.Context
 import android.os.Bundle
-import android.provider.MediaStore
+import android.provider.MediaStore.Images.Media
 import expo.modules.medialibrary.MediaLibraryConstants.ERROR_UNABLE_TO_LOAD
 import expo.modules.medialibrary.MediaLibraryConstants.ERROR_UNABLE_TO_LOAD_PERMISSION
 import org.junit.Assert.assertEquals
@@ -36,7 +36,7 @@ class GetAlbumsTests {
       arrayOf(*MockData.mockImage.toColumnArray(), bucketDisplayName),
       arrayOf(*MockData.mockVideo.toColumnArray(), bucketDisplayName)
     ))
-    cursor.setColumnNames(arrayListOf(*cursor.columnNames, MediaStore.Images.Media.BUCKET_DISPLAY_NAME))
+    cursor.setColumnNames(arrayListOf(*cursor.columnNames, Media.BUCKET_DISPLAY_NAME))
 
     val context = mockContext with mockContentResolver(cursor)
 
@@ -51,6 +51,28 @@ class GetAlbumsTests {
       assertEquals(MockData.MOCK_ALBUM_ID, firstAlbum.getString("id"))
       assertEquals(2, firstAlbum.getInt("assetCount"))
       assertEquals(bucketDisplayName, firstAlbum.getString("title"))
+    }
+  }
+
+  @Test
+  fun `GetAlbums should not list albums with null name`() {
+    //arrange
+    val bucketId = 123456
+    val bucketDisplayName = null
+
+    val cursor = mockCursor(arrayOf(
+      arrayOf(bucketId, bucketDisplayName)
+    ))
+    cursor.setColumnNames(arrayListOf(Media.BUCKET_ID, Media.BUCKET_DISPLAY_NAME))
+
+    val context = mockContext with mockContentResolver(cursor)
+
+    //act
+    TestGetAlbums(context, promise).execute()
+
+    //assert
+    promiseResolvedWithType<List<Bundle>>(promise) {
+      assertEquals(0, it.size)
     }
   }
 

